src/plugins/lib/ui/index.js

```diff
@@ -0,0 +1,222 @@
+import {Dot} from "../html/Dot.js";
+import {math} from "../../../viewer/scene/math/math.js";
+import {Marker} from "../../../viewer/scene/marker/Marker.js";
+
+export function transformToNode(from, to, vec) {
+    const fromRec = from.getBoundingClientRect();
+    const toRec = to.getBoundingClientRect();
+    vec[0] += fromRec.left - toRec.left;
+    vec[1] += fromRec.top  - toRec.top;
+};
+
+export function draggableDot3D(handleMouseEvents, handleTouchEvents, viewer, worldPos, color, ray2WorldPos, onStart, onMove, onEnd) {
```

Comment:
Could we please rename to createDraggableDot3D and use a parameter object (we try to avoid long argument lists just for style)

Reply:
Implemented by https://github.com/xeokit/xeokit-sdk/commit/72afa7852d45db04313641d956b7200cd031f23e